core/src/broadcast_stage/broadcast_duplicates_run.rs

```diff
@@ -269,44 +273,58 @@ impl BroadcastRun for BroadcastDuplicatesRun {
                 .map(|(pubkey, _)| *pubkey)
                 .collect()
         };
+
         // Broadcast data
         let cluster_nodes =
             self.cluster_nodes_cache
                 .get(slot, &root_bank, &working_bank, cluster_info);
         let socket_addr_space = cluster_info.socket_addr_space();
         let packets: Vec<_> = shreds
             .iter()
-            .filter_map(|shred| {
+            .flat_map(|shred| {
                 let seed = shred.seed(Some(self_pubkey), &root_bank);
-                let node = cluster_nodes.get_broadcast_peer(seed)?;
+                let node = cluster_nodes.get_broadcast_peer(seed);
+                if node.is_none() {
+                    return vec![];
+                }
+                let node = node.unwrap();
                 if !socket_addr_space.check(&node.tvu) {
-                    return None;
+                    return vec![];
                 }
-                if self.original_last_data_shreds.contains(&shred.signature()) {
-                    // If the node is within the partitin skip the shred.
+                if self
+                    .original_last_data_shreds
+                    .lock()
+                    .unwrap()
+                    .remove(&shred.signature())
+                {
                     if cluster_partition.contains(&node.id) {
                         info!(
                             "skipping node {} for original shred index {}, slot {}",
                             node.id,
                             shred.index(),
                             shred.slot()
                         );
-                        return None;
+                        return vec![];
                     }
+                } else if self
+                    .partition_last_data_shreds
+                    .lock()
+                    .unwrap()
+                    .remove(&shred.signature())
+                {
+                    // If the shred is part of the partition, broadcast it directly to
+                    // the partition node
```

Comment:
This is explaining what the code doing is; not why it is doing that. This comment from the change description is more helpful:
> when the partition stake is small ..., then the partition node is never selected by get_broadcast_peer()

Can you please add that instead here as the comment why `get_broadcast_peer` is ignored in this case?